base/unicode/properties.jl

```diff
@@ -0,0 +1,219 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# Unicode properties, such as General Category
+# Unix/C is* convenience functions (for now)
+
+export charprop, CharCategoryCode, UnicodeProperty, CharCategory,
+       CatLetter, CatMark, CatNumber, CatPunctuation, CatSymbol, CatSeparator, CatOther,
+       CatUpper, LowerCase, UpperCase, TitleCase, ModifierLetter, OtherLetter,
+       NonSpacingMark, CombiningMark, EnclosingMark,
+       DecimalDigit, NumericLetter, OtherNumber,
+       ConnectorPunct, DashPunct, OpenPunct, ClosePunct, BegQuotePunct, EndQuotePunct, OtherPunct,
+       MathSymbol, CurrencySymbol, ModifierSymbol, OtherSymbol,
+       SpaceSeparator, LineSeparator, ParagraphSeparator,
+       NotAssignedChar, ControlChar, FormatChar, SurrogateChar, PrivateUseChar
+
+export is_assigned_char, islower, isupper, isdigit, isalpha, isnumber, isalnum, iscntrl,
+       ispunct, isspace, isprint, isgraph
+
+# whether codepoints are valid Unicode scalar values, i.e. 0-0xd7ff, 0xe000-0x10ffff
+isvalid(::Type{Char}, ch::Unsigned) = !((ch - 0xd800 < 0x800) | (ch > 0x10ffff))
+isvalid(::Type{Char}, ch::Integer)  = isvalid(Char, Unsigned(ch))
+isvalid(::Type{Char}, ch::Char)     = isvalid(Char, UInt32(ch))
+
+isvalid(ch::Char) = isvalid(Char, ch)
+
+# Unicode General Category constants
+#=
+@enum(CharCategoryCode,
```

Comment:
I have another PR I'm am preparing, where I attempted to allow using enums that early in the build process, but so far, I haven't been able to get it working.  I've got a better approach I'll be pushing shortly.